### PR TITLE
Ct stats

### PR DESCRIPTION
**Pull request by elibritstein**

### CODE REVIEW

File: lib/netdev-rte-offloads.c [hunk 1]
```diff
@@ -4741,8 +4750,30 @@ netdev_dpdk_offload_ct_del(uint32_t mark)
 bool
 netdev_dpdk_offload_ct_active(uint32_t mark)
 {
-    VLOG_DBG("%s,%d: mark=%d", __FUNCTION__, __LINE__, mark);
-    return false;
+    struct mark_to_miss_ctx_data *data = netdev_dpdk_get_flow_miss_ctx(mark);
```
Comment (rbaryanai) [inline lib/netdev-rte-offloads.c]: you need to use find and not get, because get will create if not founb

File: lib/netdev-rte-offloads.c [hunk 2]
```diff
@@ -4741,8 +4750,30 @@ netdev_dpdk_offload_ct_del(uint32_t mark)
...
+    struct dpif_flow_stats stats;
+    ovs_u128 ctid;
+    int dir;
+    bool active = false;
+
+    if (!data) {
+        return -1;
+    }
+
+    for (active = false, dir = CT_OFFLOAD_DIR_INIT; dir < CT_OFFLOAD_NUM;
```
Comment (rbaryanai) [inline lib/netdev-rte-offloads.c]: why you don't stop on active, no need to continue the query

File: lib/conntrack.h [hunk 3]
```diff
@@ -202,7 +203,8 @@ static inline void ct_rwlock_destroy(struct ct_rwlock *lock)
     CT_TIMEOUT(OTHER_MULTIPLE, 60 * 1000) \
     CT_TIMEOUT(OTHER_BIDIR, 30 * 1000) \
     CT_TIMEOUT(ICMP_FIRST, 60 * 1000) \
-    CT_TIMEOUT(ICMP_REPLY, 30 * 1000)
+    CT_TIMEOUT(ICMP_REPLY, 30 * 1000) \
+    CT_TIMEOUT(HW_OFFLOAD, 30 * 1000)
```
[the remark this answers is not in the document]
Comment (elibritstein) [reply lib/conntrack.h]: it was for my tests. forgot to update it for the push

File: lib/netdev-rte-offloads.c (hunk 1 above)
Comment (elibritstein) [reply lib/netdev-rte-offloads.c]: ok

File: lib/netdev-rte-offloads.c (hunk 2 above)
Comment (elibritstein) [reply lib/netdev-rte-offloads.c]: ok

File: lib/netdev-rte-offloads.c [hunk 4]
```diff
@@ -1320,14 +1337,24 @@ netdev_rte_offloads_flow_stats_get(struct netdev *netdev OVS_UNUSED,
...
 
 err:
-    ovs_mutex_unlock(&ufid_to_portid_mutex);
+    ovs_mutex_unlock(mutex);
     return ret;
 }
 
+int
+netdev_rte_offloads_flow_stats_get(struct netdev *netdev OVS_UNUSED,
+                                   const ovs_u128 *ufid,
```
Comment (OphirMunk) [inline lib/netdev-rte-offloads.c]: Function netdev_rte_offloads_flow_stats_get should be static

**Pull request closed without merging**